src/fraction/approx.rs

```diff
@@ -4,51 +4,181 @@
+        /// be completely correct).
+        multiplier: BigUint,
+    },
 }
 
-impl SqrtAccuracy {
-    /// Creates a new `SqrtAccuracy` with `tail` digits after the decimal point.
-    pub fn new(tail: u32) -> SqrtAccuracy {
-        SqrtAccuracy {
-            multiplier: BigUint::from(10u8).pow(tail),
+impl Accuracy {
+    /// Returns an `Accuracy` of `n` decimal places.
+    #[must_use]
+    pub fn decimal_places<N: GenericInteger>(n: N) -> Self
+    where
+        BigUint: Pow<N>,
+        <BigUint as Pow<N>>::Output: Into<BigUint>,
+    {
+        #[cfg(feature = "lazy_static")]
+        {
+            // If we have access to pre-allocated `Accuracy` values, use them instead of allocating
+            // a new multiplier.
+            match n.to_u16() {
+                Some(20) => return Self::Dp20,
+                Some(100) => return Self::Dp100,
+                Some(500) => return Self::Dp500,
+                _ => (),
+            }
+        }
+
+        Self::Custom {
+            multiplier: Pow::pow(BigUint::from(10_u8), n).into(),
         }
     }
 
-    /// Truncates the ratio formed by `numer` and `denom` so that its decimal equivalent has only
-    /// the level of accuracy specified when `self` was created. This method takes ownership of
-    /// `numer` by necessity, but `denom` does not need to be mutated and thus only a reference to
-    /// it is required.
+    /// Returns an [`Accuracy`] of `n` digits after the point (`.`) in the representation of the
+    /// result in the given `base`.
+    ///
+    /// For example, `base_places(2, 5)` means "correct to at least 5 digits after the `.` when
+    /// printed as binary".
     ///
-    /// This does **not** simplify the return value (hence `_raw`).
-    fn truncate_ratio_raw(&self, mut numer: BigUint, denom: &BigUint) -> Ratio<BigUint> {
-        numer *= &self.multiplier;
+    /// Prefer using [`Accuracy::decimal_places`] when `base == 10`.
+    pub fn base_places<B: GenericInteger, N: GenericInteger>(base: B, n: N) -> Self
+    where
+        // Assuming `n` is anything other than really small, `base^n` will likely be pretty big, so
+        // we calculate the multiplier using `BigUint`.
+        B: Into<BigUint>,
+
+        // We need to be able to raise `BigUint(base)` to the power of `n`...
+        BigUint: Pow<N>,
+
+        // ...and get back something that we can convert straight to `BigUint`.
+        <BigUint as Pow<N>>::Output: Into<BigUint>,
+    {
+        Self::Custom {
+            multiplier: Pow::pow(base.into(), n).into(),
+        }
+    }
+
+    /// Returns a [`BigFraction`] which is equal to `fraction` according to this `Accuracy` by
+    /// "chopping off" any irrelevant digits.
+    ///
+    /// The result will be equal to `(fraction * self.multiplier()).floor() / self.multiplier()`.
+    ///
+    /// This method propagates infinity and NaN values.
+    pub fn chop<T>(&self, fraction: &GenericFraction<T>) -> BigFraction
```

Comment:
I've exposed a minimal public API for `Accuracy` so that it might be useful for users writing their own approximation code (or indeed anything else).